h2o-r/h2o-package/R/frame.R

```diff
@@ -4484,3 +4484,196 @@ h2o.stringdist <- function(x, y, method = c("lv", "lcs", "qgram", "jaccard", "jw
+#' @param y An H2OFrame object with a single numeric or binary column.
+#' @param train \code{Logical}. Whether to apply the target encoding to train data.
+#' @param blending_avg \code{Logical}. (Optional) Whether to perform blending average.
+#' @param noise_level (Optional) The amount of random noise added to the target encoding.  This helps prevent overfitting. Defaults to 0.01 * range of y.
+#' @param seed (Optional) A random seed used to generate draws from the uniform distribution for random noise. Defaults to -1.
+#' @return Returns an H2OFrame object containing the target encoding per record.
+#' @examples
+#' \donttest{
+#' library(h2o)
+#' h2o.init()
+#' 
+#' # Get Target Encoding Frame on bank-additional-full data with numeric `y`
+#' data.hex = h2o.importFile(
+#' path = "https://s3.amazonaws.com/h2o-public-test-data/smalldata/demos/bank-additional-full.csv",
+#' destination_frame = "data.hex")
+#' splits <- h2o.splitFrame(data.hex, seed = 1234)
+#' train <- splits[[1]]
+#' test <- splits[[2]]
+#' mapping <- h2o.targetencoding_map(train[c("job", "marital")], train$age)
+#' h2o.targetencoding_frame(test[c("job", "marital")], test$age, mapping, train = FALSE)
+#' 
+#' }
+#' @export
+h2o.targetencoding_frame <- function(groupby_frame, y, targetencoding_map, train, 
+                                     blending_avg = TRUE, noise_level = NULL, seed = -1){
+  
+  if (missing(groupby_frame)) 
+    stop("argument 'groupby_frame' is missing, with no default")
+  if (missing(targetencoding_map)) 
+    stop("argument 'targetencoding_map' is missing, with no default")
+  
+  if (!is.h2o(groupby_frame)) 
+    stop("argument `groupby_frame` must be a valid H2OFrame")
+  if (!is.h2o(targetencoding_map)) 
+    stop("argument `targetencoding_map` must be a valid H2OFrame")
+  
+  if (nrow(groupby_frame) != nrow(y))
+    stop("`groupby_frame` and `y` must have the same number of rows")
+  
+  if (!Reduce('&', c("numerator", "denominator") %in% colnames(targetencoding_map)))
+    stop("`targetencoding_map` must have columns: numerator, denominator")
+  
+  if (length(intersect(colnames(groupby_frame), colnames(targetencoding_map))) == 0L) 
+    stop("`groupby_frame` and `targetencoding_map` must have intersecting column names to merge on")
+  
+  if (!is.null(noise_level))
+    if (!is.numeric(noise_level) || length(noise_level) > 1L)
+      stop("`noise_level` must be a numeric vector of length 1")
+  else if (noise_level < 0)
+    stop("`noise_level` must be non-negative")
+  
+  if(!is.logical(train))
+    stop("`train` must be logical")
+  if(!is.logical(blending_avg))
+    stop("`blending_avg` must be logical")
+  
+  # Merge Target Encoding Mapping to groupby_frame
+  y_name <- colnames(y)
+  te_frame <- h2o.cbind(groupby_frame, y)
+  te_frame <- h2o.merge(te_frame, targetencoding_map, all.x = TRUE, all.y = FALSE)
+  
+  # If train = TRUE, remove value of existing row
+  if(train){
+    
```

Comment:
@brandenkmurray @mlandry22 I removed the value of the existing row before doing the blending average.  Let me know what you think of doing it in this order.